### ISSUE TITLE
exclude-rules stopped working in RC15

### ISSUE THREAD

**Issue by TWiStErRob** [arturbosch/detekt#1806]
```yaml
test-pattern: # Configure exclusions for test sources
  active: true
  patterns: # Test file regexes (make sure they're platform independent)
    - '.*[/\\]test[/\\].*'
    - '.*Test.kt'
  exclude-rules:
    - 'UnusedImports'

style:
  active: true
  UnusedImports:
    active: true
```
means that we want to check `UnusedImports` everywhere, except in test code. Test code resides in `**/test/**` folder or a file named `*Test.kt`. This was working up to and including RC14.

## Expected Behavior
```
Ruleset: style - 10min debt
        UnusedImports - [Main.kt] at exclude-rules-ignored\src\main\kotlin\pack\age\Main.kt:3:1
```

## Observed Behavior
```
Ruleset: style - 10min debt
        UnusedImports - [Main.kt] at exclude-rules-ignored\src\main\kotlin\pack\age\Main.kt:3:1
        UnusedImports - [MainTest.kt] at exclude-rules-ignored\src\test\kotlin\pack\age\MainTest.kt:3:1
```

## Steps to Reproduce
1. Clone https://github.com/TWiStErRob/repros repo
1. `cd detekt\exclude-rules-ignored`
1. `cls && gradlew clean detekt --console=plain --no-build-cache`

## Context
Can't update Gradle plugin from RC14 to RC16 because it starts to give a ton of false positives. Also people who have updated to the latest IntelliJ IDEA plugin are now seeing all-red test files, because detekt is checking rules which we don't want in test code.

## Your Environment

* Version of detekt used: RC16
* Version of Gradle used (if applicable): 5.5.1
* Operating System and version: Windows 10 x64
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/repros/tree/master/detekt/exclude-rules-ignored

**Comment by schalkms** [comment]
Thanks for reporting this problem and for the repo which reproduces this problem!

For now, you could replace the **test-pattern** with rule and rule set level excludes and includes.
The test-pattern is a deprecated feature.

**Comment by TWiStErRob** [comment]
I understand the workaround, but I don't think it's a viable solution long term, we need global configuration for tests (hopefully unit and androidTest if designed flexibly enough). We have 6 patterns for test code and excluding more than 10 rules, that's a lot of duplicate regexes.

Alternatively should we be creating 3 config files and multiple detect tasks? (If so the IDEA plugin needs serious work to support that.)

Can we update the docs with this? as this is a fully breaking change (not a deprecation) in RC15, it should be highlighted in the migration guide with emphasis.

**Comment by schalkms** [comment]
> that's a lot of duplicate regexes.

True. You duplicate the regular expressions. 
The includes and excludes are also more flexible. It’s not restricted to tests and allows to have the rules configuration fine granular. The disadvantage is the duplication of regular expressions.
We are happy that we reach version 1.0. The docs should be updated.

CC @arturbosch

**Comment by arturbosch** [comment]
Please take a look at https://arturbosch.github.io/detekt/howto-migratetestpattern.html and https://arturbosch.github.io/detekt/changelog-rc.html#migration-1.
Hopefully it will solve your migration.

Feedback and doc improvements are highly welcomed.

**Comment by robstoll** [comment]
Could you provide a list with rules and their categories? its quite cumbersome to search for the correct category for each excluded rule
